Coredns fails to compile with Go@v1.21.0

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!

If the matter is security related, please disclose it privately via security@coredns.io
-->

**What happened**:

When trying to compile from source (no changes) I receive the following error when using Go@v1.21.0
```bash
➜  coredns git:(master) ✗ make
CGO_ENABLED=0  go build -v -ldflags="-s -w -X github.com/coredns/coredns/coremain.GitCommit=9f4aa9d2-dirty" -o coredns
github.com/quic-go/quic-go/internal/handshake
# github.com/quic-go/quic-go/internal/handshake
../../go/pkg/mod/github.com/quic-go/quic-go@v0.37.3/internal/handshake/crypto_setup.go:362:37: cannot use h.allow0RTT (variable of type bool) as tls.QUICSessionTicketOptions value in argument to h.conn.SendSessionTicket
make: *** [Makefile:16: coredns] Error 1
```

**What you expected to happen**:

The compile to succeed

**How to reproduce it (as minimally and precisely as possible)**:

1. install [Go@v1.21.0 ](https://go.dev/dl/)
2. clone the repo `git clone https://github.com/coredns/coredns.git`
3. Run make

**Anything else we need to know?**:

When downgrading to `Go@v1.20.7` coredns compiled as expected.

**Environment**:

- the version of CoreDNS: `master`

- OS (e.g: `cat /etc/os-release`):
    - `Ubuntu 22.10`
    - and `Ubuntu 22.04.3`

--- a/go.mod
+++ b/go.mod
@@ -24,7 +24,7 @@ require (
 	github.com/prometheus/client_golang v1.16.0
 	github.com/prometheus/client_model v0.4.0
 	github.com/prometheus/common v0.44.0
-	github.com/quic-go/quic-go v0.37.3
+	github.com/quic-go/quic-go v0.37.4
 	go.etcd.io/etcd/api/v3 v3.5.9
 	go.etcd.io/etcd/client/v3 v3.5.9
 	golang.org/x/crypto v0.12.0

